packages/orbit-components/src/Card/CardSection/index.tsx

```diff
@@ -26,108 +23,102 @@ const CardSection = ({
   onExpand,
   dataTest,
   actions,
-  noSeparator,
-}: Props) => {
-  const { addSection, removeSection, index, roundedBorders, noBorderTop, isOpened } = useCard();
-  const [opened, setOpened] = React.useState(isOpened || initialExpanded);
+}: Props) {
+  const [opened, setOpened] = React.useState(initialExpanded);
 
   const isControlled = expanded != null;
 
-  // effect that solves controlled component
   React.useEffect(() => {
     if (isControlled) {
-      if (expanded) {
-        addSection(index);
-        setOpened(true);
-      } else {
-        removeSection(index);
-        setOpened(false);
-      }
+      setOpened(expanded);
     }
-  }, [addSection, expanded, index, isControlled, removeSection]);
+  }, [isControlled, expanded]);
 
-  // effect that solves initialExpanded behavior
-  React.useEffect(() => {
-    if (initialExpanded) {
-      addSection(index);
-      setOpened(true);
-    }
-  }, [addSection, index, initialExpanded]);
-
-  const handleClick = () => {
+  function handleClick() {
     if (!isControlled) {
-      if (!opened) {
-        addSection(index);
-        setOpened(true);
-      } else {
-        removeSection(index);
-        setOpened(false);
-      }
-    }
-
-    if (opened && onClose) {
-      onClose();
-    }
-
-    if (!opened && onExpand) {
-      onExpand();
-    }
-  };
-
-  const handleKeyDown = (ev: React.KeyboardEvent<HTMLDivElement>) => {
-    if (ev.keyCode === KEY_CODE_MAP.SPACE) {
-      ev.preventDefault();
+      setOpened(state => !state);
     }
 
-    if (ev.keyCode === KEY_CODE_MAP.ENTER || ev.keyCode === KEY_CODE_MAP.SPACE) {
-      handleClick();
+    if (opened) {
+      onClose?.();
+    } else {
+      onExpand?.();
     }
-  };
+  }
 
   const slideID = useRandomId();
 
   return (
-    <CardWrapper
-      dataTest={dataTest}
-      roundedTop={roundedBorders.top}
-      roundedBottom={roundedBorders.bottom}
-      expanded={opened}
+    // Needs to capture bubbled click events from the <button> below
+    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
+    <div
+      className={cx(
+        "duration-fast lm:border-x border-b transition-all ease-in-out",
+        opened && "my-xs rounded-normal shadow-action-active [&+*]:border-t",
+        onClick != null && "hover:bg-white-normal-hover cursor-pointer",
+      )}
+      data-test={dataTest}
+      role={onClick == null ? undefined : "button"}
+      tabIndex={onClick == null ? undefined : -1}
```

Comment:
Sorry, do not want to block and bother again, but this seems broken 😞  It's having onClick it will have tabIndex -1? I guess it should be focusable, otherwise, tab navigation is broken (in master it's focusable and works correctly)

Reply:
https://kiwicom.github.io/orbit/?path=/story/card--card-with-sections can you focus the first one?

Reply:
ah okay I originally thought it is jsut for event bubbling. sec I will adjust

Reply:
the value of tabIndex should be `0`, so the order is natural ;)

Reply:
also still missing the onKeyDown: 

```
onKeyDown={handleKeyDown(onClick)}
```

the same as in [Tile component](https://github.com/kiwicom/orbit/blob/master/packages/orbit-components/src/Tile/index.tsx#L59) ;)

Reply:
fixed